### ISSUE TITLE
NullPointerException when auto-deleting sent filled forms

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#2709]
#### Software and hardware versions 
Collect v1.18.0, Android v7.0

#### Problem description
After an auto-send, there's a crash when deleting the filled form.

[Play Store report](https://play.google.com/apps/publish/?account=6923365842552008664#AndroidMetricsErrorsPlace:p=org.odk.collect.android&appVersion=3062&clusterName=apps/org.odk.collect.android/clusters/de91a609&detailsAppVersion=3062&detailsSpan=7) (needs authentication):
```
java.lang.NullPointerException: 
  at org.odk.collect.android.provider.InstanceProvider.deleteAllFilesInDirectory (InstanceProvider.java:227)
  at org.odk.collect.android.provider.InstanceProvider.delete (InstanceProvider.java:288)
  at android.content.ContentProvider$Transport.delete (ContentProvider.java:374)
  at android.content.ContentResolver.delete (ContentResolver.java:1422)
  at org.odk.collect.android.upload.AutoSendWorker.doWork (AutoSendWorker.java:160)
  at androidx.work.Worker$1.run (Worker.java:57)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1133)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:761)
```

#### Steps to reproduce the problem
Unknown. One possibility since it happens on Android 7 is that the permissions are revoked before the deletion happens. I'm not sure how that would happen but seems possible. 

#### Expected behavior
No crash.

#### Other information 
I am thinking of checking for null and if there is a null, logging an exception that checks the granted permissions.

**Comment by lognaturel** [comment]
On further analysis, I really don't think this can be a permissions problem since `delete` does the check right before.

I hesitate to add a null check because this has only happened once so far and seeing if it happens more might give us some hints as to what's going on. Unless anyone else has any ideas, I'll close this and re-open only if it keeps happening.

**Comment by grzesiek2010** [comment]
From my experience, I know that sometimes `listFiles()` method might return null probably it's a random issue really rare. We have already fixed some similar problems just adding a null check. I wanted to do that the same here but I've just noticed you have a different idea. I don't think putting that on hold helps because as I said it's a random issue (I mean there is nothing more we can do here) and probably we should just add that null check.

**Comment by grzesiek2010** [comment]
If you agree you can just reopen my pr #2721

**Comment by shobhitagarwal1612** [comment]
> I know that sometimes listFiles() method might return null probably it's a random issue really rare

I have noticed this when storage permission is not granted.

**Comment by shobhitagarwal1612** [comment]
I encountered it while working on PR https://github.com/opendatakit/collect/pull/2697 and have fixed the issue in that PR

**Comment by lognaturel** [comment]
Thanks, @shobhitagarwal1612! That PR makes pretty significant behavior changes that have not been discussed so it’s not a candidate for a point release. If you have the time, please issue a PR to narrowly fix this crash. If you don’t, I can do it during my day.

**Comment by grzesiek2010** [comment]
> I have noticed this when storage permission is not granted.

I'm not sure if it's related. I saw the same issue a long time ago before we started using new permissions. Additionally, in all other cases, it crashes when the permission is not granted so why not here.

**Comment by lognaturel** [comment]
[Here](https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/5bd8bc81f8b88c2963df7a42?time=last-seven-days&sessionId=5BD9AABE024C00011ABFE163D8DF4123_DNE_0_v2) (requires Crashlytics credentials) is a report from an Android 5 device. While [`listFiles` does seem to return null without permissions](https://stackoverflow.com/questions/42133758/file-listfiles-returns-null-when-it-shouldnt?rq=1), I think it's vanishingly unlikely to be permissions related in this case.

Let's do nothing for now so we keep getting logs. If a user reports the crash, that would also be helpful.
